examples/mlp.py

```diff
@@ -110,15 +129,16 @@ def cost(model, x, target):
     diff = target - y
     return sum(diff * diff)
 
+f32 = Float[32]
 
-# @myia(backend_options={'target': device_type})
-@myia(backend='pytorch', backend_options={'device': device_type})
+@myia(backend=backend, backend_options=backend_options, return_backend=True)
 def step(model, x, y):
     """Returns the loss and parameter gradients."""
     # value_and_grad will return cost(model, x, y) and dcost(...)/dmodel.
     # The 'model' argument can be omitted: by default the derivative wrt
     # the first argument is returned.
-    return value_and_grad(cost, 'model')(model, x, y)
+    cost, dmodel = value_and_grad(cost, 'model')(model, x, y)
+    return cost, model - dmodel * scalar_cast(0.01, f32)
```

Comment:
The dtype is not necessarily f32 (see the `dtype = 'float32'` line). `myia.dtype.DTYPE_TO_MTYPE[dtype]` should return the correct type to cast to, although you might need to assign that to a global variable outside of the step function.